packages/perseus/src/widgets/number-line/score-number-line.ts

```diff
@@ -1,50 +1,36 @@
 import {number as knumber} from "@khanacademy/kmath";
 
+import validateNumberLine from "./validate-number-line";
+
 import type {PerseusScore} from "../../types";
 import type {
-    PerseusNumberLineRubric,
+    PerseusNumberLineScoringData,
     PerseusNumberLineUserInput,
 } from "../../validation.types";
 
 function scoreNumberLine(
-    state: PerseusNumberLineUserInput,
-    rubric: PerseusNumberLineRubric,
+    userInput: PerseusNumberLineUserInput,
+    scoringData: PerseusNumberLineScoringData,
 ): PerseusScore {
-    const range = rubric.range;
-    const divisionRange = state.divisionRange;
-    const start = rubric.initialX != null ? rubric.initialX : range[0];
-    const startRel = rubric.isInequality ? "ge" : "eq";
-    const correctRel = rubric.correctRel || "eq";
+    const validationError = validateNumberLine(userInput, scoringData);
+    if (validationError) {
+        return validationError;
+    }
+
+    const correctRel = scoringData.correctRel || "eq";
     const correctPos = knumber.equal(
-        state.numLinePosition,
-        rubric.correctX || 0,
+        userInput.numLinePosition,
+        scoringData.correctX || 0,
     );
-    const outsideAllowedRange =
-        state.numDivisions > divisionRange[1] ||
-        state.numDivisions < divisionRange[0];
 
-    // TODO: I don't think isTickCrtl is a thing anymore
-    if (state.isTickCrtl && outsideAllowedRange) {
```

Comment:
I'm guessing when exercises allow users to set their own interval, then this check was necessary in case they put in an invalid number of divisions.

I checked the number line editor, and it still has a checkbox for `isTickCrtl` and the json still has the field (though it defaults to not being there). I tried removing `state.isTickCrtl` from line 27 and all the tests still pass. It doesn't look like it is referenced anywhere other than tests and this line here. Wouldn't this always be false if `isTickCrtl` is not in use? Maybe this validation check hasn't been working?

Or maybe this check is not necessary if we are no longer allowing the user to set their own tick interval. It _is_ possible to turn this to `true` in the editor, so I think we should only remove it if that checkbox ever gets removed, just in case it has been set to true in the past or in case anyone sets it to true in the future.

I don't think this property should be on userInput though. Should we move it to validationData?

Reply:
Maybe moving it can be a later fix if needed. Just left it for now as the only logic moved into the validation function.